fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/scheduler/SmsMessageScheduledJobServiceImpl.java

```diff
@@ -118,18 +122,26 @@ public void sendMessagesToGateway() {
                     Collection<SmsMessageApiQueueResourceData> apiQueueResourceDatas = new ArrayList<>();
                     while (pendingMessageIterator.hasNext()) {
                         SmsMessage smsData = pendingMessageIterator.next();
-
-                        SmsMessageApiQueueResourceData apiQueueResourceData = SmsMessageApiQueueResourceData.instance(smsData.getId(),
-                                tenantIdentifier, null, null, smsData.getMobileNo(), smsData.getMessage(), smsData.getSmsCampaign()
-                                        .getProviderId());
-                        apiQueueResourceDatas.add(apiQueueResourceData);
-                        smsData.setStatusType(SmsMessageStatusType.WAITING_FOR_DELIVERY_REPORT.getValue());
-                        toSaveMessages.add(smsData) ;
+                        if(smsData.isNotification()){
+                        	smsData.setStatusType(SmsMessageStatusType.WAITING_FOR_DELIVERY_REPORT.getValue());
```

Comment:
What is the status lifecycle of SmsMessage? Is it the same for notifications as well? Is it the same for GCM and FCM?